src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -192,7 +192,7 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                 If startingStatementInfo.MatchingBlockConstruct IsNot Nothing Then
                     ' If we're expanding backwards because of editing an end construct, we don't wan to run
                     ' expensive semantic formatting checks.  We really just want to fix up indentation.
-                    formattingInfo.UseSemantics = finalSpanStart <= startingStatementInfo.MatchingBlockConstruct.SpanStart
+                    formattingInfo.UseSemantics = (finalSpanStart <= startingStatementInfo.MatchingBlockConstruct.SpanStart) OrElse startingStatementInfo.MatchingBlockConstruct.Parent.IsExecutableBlock
```

Comment:
i don't really think we want to do this.  this still has the issue mentioned above, namely that we might use semantics on an enormous block of code :)

i think a better approach would be to simply statethat if you're editing one of these lines we use semantics on htat line alone, and syntax on the entire span.

Reply:
Are you saying to do 2 formats? First on the line line then another on the block. This function does not do the actual formatting its parent does and the parent already handles large blocks.

Reply:
i'm saying that the solution i would want would still keep the syntactic formatting for the larger range (i.e. the entire `if` lbock), but only do the semantic formatting for the statement we're on (the single `elseif` statement). 

That seems to give the best of both worlds here, without potentially regressing perf a lot on large constructs.  

Thanks!